pkg/infrastructure/nutanix/clusterapi/clusterapi.go

```diff
@@ -146,44 +147,86 @@ func (p Provider) Ignition(ctx context.Context, in infracapi.IgnitionInput) ([]b
 	}
 	imgReq.Metadata = imgMeta
 
+	retries := 5
+	completed := false
+	for i := 0; !completed && i < retries; i++ {
```

Comment:
Since you're making API calls to upload/delete content, what about using https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoffWithContext which will sleep in between attempts and will stop the operation if the context is cancelled or times out?

Reply:
Changed to use wait#ExponentialBackoffWithContext